catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswSubscriptionEndpoint.java

```diff
@@ -0,0 +1,546 @@
+import org.codice.ddf.spatial.ogc.csw.catalog.endpoint.event.CswSubscription;
+import org.codice.ddf.spatial.ogc.csw.catalog.endpoint.event.CswSubscriptionConfigFactory;
+import org.codice.ddf.spatial.ogc.csw.catalog.transformer.TransformerManager;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.event.Subscription;
+import ddf.catalog.operation.QueryRequest;
+import net.opengis.cat.csw.v_2_0_2.AcknowledgementType;
+import net.opengis.cat.csw.v_2_0_2.EchoedRequestType;
+import net.opengis.cat.csw.v_2_0_2.GetRecordsType;
+import net.opengis.cat.csw.v_2_0_2.ObjectFactory;
+import net.opengis.cat.csw.v_2_0_2.QueryType;
+
+public class CswSubscriptionEndpoint implements Subscribe {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CswSubscriptionEndpoint.class);
+
+    private Map<String, ServiceRegistration<Subscription>> registeredSubscriptions =
+            new ConcurrentHashMap<>();
+
+    private ObjectFactory objectFactory = new ObjectFactory();
+
+    private static final String UUID_URN = "urn:uuid:";
+
+    private DatatypeFactory datatypeFactory;
+
+    private BundleContext context;
+
+    private final TransformerManager schemaTransformerManager;
+
+    private final TransformerManager mimeTypeTransformerManager;
+
+    private Validator validator;
+
+    private CswQueryFactory queryFactory;
+
+    public CswSubscriptionEndpoint(BundleContext context,
+            TransformerManager mimeTypeTransformerManager,
+            TransformerManager schemaTransformerManager, Validator validator,
+            CswQueryFactory queryFactory) {
+        this.context = context;
+        this.mimeTypeTransformerManager = mimeTypeTransformerManager;
+        this.schemaTransformerManager = schemaTransformerManager;
+        this.validator = validator;
+        this.queryFactory = queryFactory;
+
+        try {
+            this.datatypeFactory = DatatypeFactory.newInstance();
+        } catch (DatatypeConfigurationException e) {
+            LOGGER.error("Error initializing datatypeFactory", e);
+        }
+    }
+
+    /**
+     * Deletes an active subscription
+     *
+     * @param requestId the requestId of the subscription to be removed
+     * @return Acknowledgment   returns a CSW Acknowledgment message with the subscription that was
+     * deleted or an empty 404 if none are found
+     * @throws CswException
+     */
+    @Override
+    @DELETE
+    @Path("/{requestId}")
+    @Produces({MediaType.WILDCARD})
+    public Response deleteRecordsSubscription(@PathParam("requestId") String requestId)
+            throws CswException {
+        CswSubscription subscription = getSubscription(requestId);
```

Comment:
There is a synchronization "gap" here where we access the map to check if it contains something and then operate on it, all outside of a synchronization block. As I mentioned in my comment below, we probably want to revisit how this is being done to avoid potential concurrency problems.